modules/javafx.graphics/src/main/java/javafx/animation/Animation.java

```diff
@@ -124,6 +122,24 @@
 
     private static final double EPSILON = 1e-12;
 
+    /**
+     * Checks if the rate is effectively 0.
+     * @param rate
+     * @return true i.f.f. abs(rate) < EPSILON
+     */
+    static final boolean isNearZero(double rate) {
+        return Math.abs(rate) < EPSILON;
+    }
+
+    /**
+     * Checks if 2 rates are effectively equal.
+     * @param rateMagnitude
+     * @return true i.f.f. Math.abs(rate1 - rate2) < EPSILON
+     */
+    private static boolean areNearEqual(double rate1, double rate2) {
+        return isNearZero(rate2 - rate1);
+    }
+
```

Comment:
I agree that these methods are better suited there, but I'm not sure the same epsilon value will be suitable for other places that will want to use these. That value is somewhat arbitrary anyway I think.

Reply:
Either is fine with me.

Reply:
> but I'm not sure the same epsilon value will be suitable for other places that will want to use these

True, EPSILON is defined in some other classes with different value. Only Animation classes define EPSILON as 1e-12. So it may not be reasonable to include Animation specific helper method in common Util class. These methods would be more suitable for Animation only util class. I think I am good with the current way.